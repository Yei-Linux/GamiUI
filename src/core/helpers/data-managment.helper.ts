export const mapEnum = (enumerable: any): any[] => {
  let enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  );
  return enumMembers;
};

