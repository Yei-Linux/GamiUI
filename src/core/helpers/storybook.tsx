import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};