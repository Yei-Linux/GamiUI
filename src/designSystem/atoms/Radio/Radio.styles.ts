import { css } from "@emotion/react";
import styled from "@emotion/styled";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const RadioGroupWrapper = styled.div<{
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  textAlign?: string;
  fontWeight?: string;
}>`
  margin: 0;
  padding: 0;
  font-size: 14px;

  ${(props: any) => twinStyles(props)};
`;

export const RadioContainer = styled.label`
  padding: 0;
  position: relative;
  display: inline-flex;
  margin: 0 8px 0 0;
  cursor: pointer;
`;

export const RadioCheck = styled.span`
  margin: 0;
  padding: 0;
  position: relative;
  display: inline-block;
  outline: none;
  cursor: pointer;
`;

export const RadioInput = styled.input`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1;
  cursor: pointer;
  opacity: 0;
`;

export const RadioInner = styled.span<{ isChecked: boolean }>`
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 16px;
  height: 16px;
  background-color: #fff;
  border: 1px solid #d9d9d9;
  border-radius: 50%;
  transition: all 0.3s;

  ${(props) =>
    props.isChecked &&
    css`
      border-color: #1890ff;
    `}

  &::after {
    position: absolute;
    top: 3px;
    left: 3px;
    display: table;
    width: 8px;
    height: 8px;
    background-color: #1890ff;
    border-top: 0;
    border-left: 0;
    border-radius: 8px;
    transform: scale(0);
    opacity: 0;
    transition: all 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
    content: " ";

    ${(props) =>
      props.isChecked &&
      css`
        transform: scale(1);
        opacity: 1;
      `}
  }
`;

export const RadioText = styled.span`
  margin: 0 10px;
`;
