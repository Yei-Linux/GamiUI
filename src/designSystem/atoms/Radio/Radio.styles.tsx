import styled from '@emotion/styled'
import {
  BorderType,
  HeightType,
  ShadowType,
  WidthType,
} from 'core/domain/types'
import { opacity, spacing, zIndex } from 'styles/tokens'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'

export const RadioGroup = styled.div<{
  $border?: BorderType
  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
}>`
  margin: ${spacing.margin.none};
  padding: ${spacing.padding.none};
  font-size: 14px;

  ${({ $border, $shadow, $width, $heigth }) =>
    setGenericPropStyles({
      border: $border || 'none',
      shadow: $shadow || 'none',
      width: $width || 'auto',
      heigth: $heigth || 'auto',
    })};
`

export const RadioItem = styled.label`
  margin: 0 8px 0 0;
  padding: ${spacing.padding.none};

  position: relative;
  display: inline-flex;

  &:hover {
    cursor: pointer;
  }
`

export const RadioCheck = styled.span`
  margin: ${spacing.margin.none};
  padding: ${spacing.padding.none};

  position: relative;
  display: inline-block;

  outline: none;

  &:hover {
    cursor: pointer;
  }
`

export const RadioInput = styled.input`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  opacity: ${opacity[0]};
  z-index: ${zIndex[0]};

  &:hover {
    cursor: pointer;
  }
`

export const RadioInner = styled.span`
  position: relative;
  top: 0;
  left: 0;
  display: block;
  width: 16px;
  height: 16px;
  background-color: #fff;
  border: 1px solid #d9d9d9;
  border-radius: 50%;
  transition: all 0.3s;

  &.checked {
    border-color: #1890ff;

    &::after {
      transform: scale(1);
      opacity: 1;
    }
  }

  &::after {
    display: table;
    position: absolute;
    top: 3px;
    left: 3px;

    width: 8px;
    height: 8px;

    background-color: #1890ff;
    border-top: 0;
    border-left: 0;
    border-radius: 8px;
    opacity: 0;

    transform: scale(0);
    transition: all 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
    content: ' ';
  }
`

export const RadioText = styled.span`
  margin: 0 10px;
`
