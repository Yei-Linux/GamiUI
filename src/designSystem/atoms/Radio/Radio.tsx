import React, { useContext, useEffect } from "react";
import RadioProvider from "../../../context/RadioProvider";
import context from "../../../context/RadioProvider/context";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import {
  RadioCheck,
  RadioContainer,
  RadioGroupWrapper,
  RadioInner,
  RadioInput,
  RadioText,
} from "./Radio.styles";

interface IRadioItem {
  children: string;
  value: string;
  isChecked?: boolean;
}

export interface IRadio extends IGeneralProps {
  /**
   * Function to detect changes
   */
  onChangeFormItem?: any;
  /**
   * Children Element
   */
  children: React.ReactNode[];
  /**
   * Function to detect changes
   */
  name?: any;
  /**
   * Function to detect changes
   */
  value?: any;
}

const RadioContent = ({ children, onChangeFormItem, ...args }: IRadio) => {
  const { radioSelected } = useContext(context);

  useEffect(() => {
    onChangeFormItem(radioSelected);
  }, [radioSelected]);

  return <RadioGroupWrapper {...args}>{children}</RadioGroupWrapper>;
};

const Radio = ({ children, ...args }: IRadio) => {
  return (
    <RadioProvider>
      <RadioContent {...args}>{children}</RadioContent>
    </RadioProvider>
  );
};

Radio.Item = ({ children, value, isChecked = false }: IRadioItem) => {
  const { radioSelected, setRadioSelected } = useContext(context);

  useEffect(() => {
    isChecked && handleRadioChange();
  }, []);

  const handleRadioChange = () => setRadioSelected({ text: children, value });

  return (
    <RadioContainer>
      <RadioCheck>
        <RadioInput onClick={handleRadioChange} />
        <RadioInner isChecked={radioSelected?.value == value} />
      </RadioCheck>
      <RadioText>{children}</RadioText>
    </RadioContainer>
  );
};

Radio.defaultProps = {
  width: "NORMAL",
  heigth: "SMALL",
  border: "MEDIUM",
};

export default Radio;
