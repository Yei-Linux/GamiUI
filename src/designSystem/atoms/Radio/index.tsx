export { default } from "./Radio";
export { options } from "./constants";
