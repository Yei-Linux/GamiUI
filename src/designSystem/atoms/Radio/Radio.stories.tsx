import React from "react";
import { Meta } from "@storybook/react";
import { stylesControl } from "../../../core/helpers/storybook";

import Radio from ".";

export default {
  title: "Atoms/Radio",
  component: Radio,
  args: { ...Radio.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const Basic = (args: any) => {
  return (
    <Radio {...args}>
      <Radio.Item isChecked value="one">
        A
      </Radio.Item>
      <Radio.Item value="two">B</Radio.Item>
      <Radio.Item value="three">C</Radio.Item>
    </Radio>
  );
};
Basic.args = {
  width: "NORMAL",
  heigth: "SMALL",
  shadow: "",
  border: "MEDIUM",
};
