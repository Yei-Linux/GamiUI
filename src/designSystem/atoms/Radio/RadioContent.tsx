import React, { useEffect } from 'react'
import useRadioStore from 'hooks/store/useRadioStore'
import { IRadio } from './Radio'
import * as S from "./Radio.styles"

const RadioContent = ({
  children,
  onChangeFormItem,
  ...generalInputProps
}: IRadio) => {
  const { radioSelected } = useRadioStore()

  useEffect(() => {
    onChangeFormItem?.(radioSelected)
  }, [radioSelected])

  return <S.RadioGroup {...generalInputProps}>{children}</S.RadioGroup>
}

export default RadioContent