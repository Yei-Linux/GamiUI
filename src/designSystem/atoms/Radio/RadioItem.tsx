import classNames from 'classnames'
import useRadioStore from 'hooks/store/useRadioStore'
import React, { useEffect } from 'react'
import * as S from './Radio.styles'

export interface IRadioItem {
  /**
   * Radio Item content
   */
  children: string
  /**
   * Radio Value
   */
  value: string
  /**
   * Is checked prop
   */
  isChecked?: boolean
}

const RadioItem = ({ children, value, isChecked = false }: IRadioItem) => {
  const { radioSelected, setRadioSelected } = useRadioStore()

  useEffect(() => {
    isChecked && handleRadioChange()
  }, [])

  const handleRadioChange = () => setRadioSelected({ text: children, value })

  return (
    <S.RadioItem>
      <S.RadioCheck>
        <S.RadioInput onClick={handleRadioChange} />
        <S.RadioInner
          className={classNames({ checked: radioSelected?.value == value })}
        />
      </S.RadioCheck>
      <S.RadioText>{children}</S.RadioText>
    </S.RadioItem>
  )
}

export default RadioItem
