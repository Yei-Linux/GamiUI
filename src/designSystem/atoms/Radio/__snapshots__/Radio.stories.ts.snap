// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Radio Basic 1`] = `
<div
  className="css-7taf9n-RadioGroupWrapper eldjpxi0"
  fontWeight="BOLD"
  width="NORMAL"
>
  <label
    className="css-1ur0gxt-RadioContainer eldjpxi1"
  >
    <span
      className="css-1c1trwd-RadioCheck eldjpxi2"
    >
      <input
        className="css-uckndt-RadioInput eldjpxi3"
        onClick={[Function]}
      />
      <span
        className="css-egwf0x-RadioInner eldjpxi4"
      />
    </span>
    <span
      className="css-26doda-RadioText eldjpxi5"
    >
      A
    </span>
  </label>
  <label
    className="css-1ur0gxt-RadioContainer eldjpxi1"
  >
    <span
      className="css-1c1trwd-RadioCheck eldjpxi2"
    >
      <input
        className="css-uckndt-RadioInput eldjpxi3"
        onClick={[Function]}
      />
      <span
        className="css-egwf0x-RadioInner eldjpxi4"
      />
    </span>
    <span
      className="css-26doda-RadioText eldjpxi5"
    >
      B
    </span>
  </label>
  <label
    className="css-1ur0gxt-RadioContainer eldjpxi1"
  >
    <span
      className="css-1c1trwd-RadioCheck eldjpxi2"
    >
      <input
        className="css-uckndt-RadioInput eldjpxi3"
        onClick={[Function]}
      />
      <span
        className="css-egwf0x-RadioInner eldjpxi4"
      />
    </span>
    <span
      className="css-26doda-RadioText eldjpxi5"
    >
      C
    </span>
  </label>
</div>
`;
